lib/atoi/a2i.h

```diff
@@ -15,51 +15,249 @@
 #include "attr.h"
 
 
-#define a2i(TYPE, ...)                                                        \
+#define a2i(TYPE, n, s, ...)                                                  \
```

Comment:
It would be nice to explain in comments what all these macros do

Reply:
[liba2i-0.8-dirty.pdf](https://github.com/shadow-maint/shadow/files/15236155/liba2i-0.8-dirty.pdf)

I could paste the manual page in a comment, if you want.

Reply:
In reality the comment would be, what added value do these macros bring? I assume they are a wrapper on top of all `a2*` functions and they can be used for any data type as they abstract the data type (to a level). Explaining that will help people understand what they do, and what they can be used for. This way, we'll get more developers using them.

Reply:
Ahh, I thought you were asking for documentation about the entire functionality.  For what the macros do on top of the function, there's a small subsection in the manual page:
<file:///home/alx/Downloads/liba2i-0.8-dirty-5.pdf#pdf:bm11>

But it only documents what it does, and not why.  Since I followed what C23 has done with strchr(3) et al., you could check why ISO C23 did that.  The following are interesting to read:
<https://thephd.dev/c23-is-coming-here-is-what-is-on-the-menu#n3020---qualifier-preserving-standard-functions>
<https://www.open-std.org/jtc1/sc22/wg14/www/docs/n3020.pdf>

Reply:
Wait, are those macros provided by liba2i or shadow?

Reply:
provided by shadow.

liba2i is a library I wrote _after_ writing the same macros for shadow, to be able to use them in other projects.

At the moment, shadow has no relation with liba2i, except that I write the code for both projects.

Reply:
My plan is to first put all this code in shadow, and in a year or two replace those by a dependency on liba2i.

Reply:
ok, then can you add a link as a comment in the code to the documentation that is already written for the liba2i project? I don't think it's necessary to copy all that information in this project, but it's interesting for other developers to have a quick access to the documentation

Reply:
Done.